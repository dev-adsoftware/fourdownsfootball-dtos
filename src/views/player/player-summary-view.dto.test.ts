import { PlayerSummaryView } from '.';

describe('given: player summary data', () => {
  describe('when: I create a view', () => {
    it('then: PlayerSummaryView was returned', async () => {
      const view = new PlayerSummaryView().init({
        id: 'jest.player.id',
        sequence: '0',
        firstName: 'jest.firstname',
        lastName: 'jest.lastname',
        suffix: '',
        teamId: 'jest.team.id',
        kicking: 50,
      });
      expect(view).toEqual({
        id: 'jest.player.id',
        sequence: '0',
        firstName: 'jest.firstname',
        lastName: 'jest.lastname',
        suffix: '',
        teamId: 'jest.team.id',
        kicking: 50,
      });
    });
  });
});
