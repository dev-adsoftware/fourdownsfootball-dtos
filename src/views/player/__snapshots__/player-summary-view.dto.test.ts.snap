// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: missing data when: I initialize then: error matched snapshot 1`] = `[Error: PlayerSummaryView: id must be a string, sequence must be a number string, teamId must be a string, attributes must be an object, statAttributes must be an object]`;

exports[`given: missing nested data when: I initialize then: error matched snapshot 1`] = `
[Error: PlayerSummaryView: id must be a string, sequence must be a number string, teamId must be a string, An instance of an object has failed the validation:
 - property attributes.firstName has failed the following constraints: isString 
 - property attributes.lastName has failed the following constraints: isString 
 - property attributes.age has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.height has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.weight has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.jerseyNumber has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.speed has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.strength has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.agility has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.quickness has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.passingStrength has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.shortPassAccuracy has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.mediumPassAccuracy has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.longPassAccuracy has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.scramblingPassAccuracy has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.scrambling has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.ballHandling has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.rushing has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.breakingTackles has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.routeRunning has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.catching has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.spectacularCatching has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.catchingInTraffic has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.jumping has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.passBlocking has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.runBlocking has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.runStuffing has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.passRushing has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.zoneCovering has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.manCovering has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.awareness has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.intelligence has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.clutch has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.endurance has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.stamina has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.fragility has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.toughness has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.returning has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.kickingPower has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.kickingAccuracy has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.puntingPower has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.puntingAccuracy has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.aggressiveness has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.riskTaking has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.sportsmanship has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.coachability has failed the following constraints: max, min, isDivisibleBy, isNumber 
 - property attributes.leadership has failed the following constraints: max, min, isDivisibleBy, isNumber 
, An instance of an object has failed the validation:
 - property statAttributes.plays has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.kickoffs has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.kickoffYds has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.kickoffTouchbacks has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.kickReturns has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.kickReturnYds has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.kickReturnTDs has failed the following constraints: isDivisibleBy, isNumber 
 - property statAttributes.tackles has failed the following constraints: isDivisibleBy, isNumber 
]
`;
